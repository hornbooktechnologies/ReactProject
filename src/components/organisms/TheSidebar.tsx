  import React, { useState, useEffect, Fragment } from 'react';
import { localStorageUtils } from "../../utils/LocalStorageUtils";
import { NavigationPanel } from "./NavigationPanel";
import { Link } from "react-router-dom";
import { URLConstants, NavConstant } from "../../utils/AppConstants";
import "./../../assets/scss/sidebar.scss";
import { cssClassName } from "../../utils/CssConstants";
const TheSidebar = () => {
  const [navigation, setNavigation] = useState(NavigationPanel.getOverviewNav());
  const [ukShow, setUkShow] = useState(true);
  const [subnav, setSubnav] = useState();
  const [subchildpath, setSubChildPath] = useState(
        window.location.pathname === "/"
      ? URLConstants.ROUTE_CURRENT_STATUS_PATH
      : window.location.pathname
  );
  const [pathName, setPathName] = useState(window.location.pathname === "/"
  ? URLConstants.ROUTE_CURRENT_STATUS_PATH
  : window.location.pathname);

  useEffect((): void => {
    
    let userPreferences = localStorageUtils.getSelectedPlantUserPreferance();
    if (userPreferences) {
      let dynamicNavigation = NavigationPanel.generateNavigation(userPreferences);
      setNavigation(dynamicNavigation);
      dynamicNavigation.map((items)=>(
        items._children.map((children: { to: string; },index: any)=>(
       ( children.to === window.location.pathname) && 
    (
          localStorageUtils.setSelectedPlantUnitId(items && items.unitId && items.unitId.slice(0,3)),
          localStorageUtils.setSelectedUnit(items.unitname)
    )
        ))
      ))  
    }
  }, [subchildpath,ukShow]);
    
  const onSideDropDownClick = (items: any, index: number, e: any) => {
     if (e.target.id === NavConstant.SUBITEMID) {
      setUkShow(true);
    } else if (subnav === index) {
      setUkShow(!ukShow);
    } else {
      setUkShow(true);
    }
    setSubnav(index);
    setPathName("")
  };

  const handlesubItemClick = (children: { to: React.SetStateAction<string>; },i: number) => {
   setSubChildPath(children.to);
  }

  return (
    <>
     <div className="uk-sidebar sidebar-left">
       <ul
          className="uk-sidebar-nav h-100  uk-ps"
          uk-scrollspy="closest: li; scroll: true"
        >
          {navigation.map((items: any, index:number) => {
           return (
              <Fragment key={index}>
                <li
                  className={`uk-sidebar-nav-item uk-c-pointer uk-parent ${
                    items._children &&
                    `uk-sidebar-nav-dropdown ${
                      (subnav === index  || items._children.filter(element => element.to === pathName).length > 0 ) && ukShow ? "uk-show" : ""
                    }`
                  }`}
                  key={index.toString()}
                  onClick={(e) => {
                    onSideDropDownClick(items, index, e);
                  }}
                >
                  <a
                    href="false"
                    className="uk-sidebar-nav-link1"
                    uk-tooltip={`title:${items.name}; pos:left;`}
                    tabIndex={0}
                    onClick={(e)=> e.preventDefault()}
                    >
                    <i className={`${items.icon}`}></i>
                    <span>{items.name}</span>
                    <i className={`${cssClassName.JERA_ARROW_DOWN_1}`}></i>
                  </a>
                  {items._children && (
                    <ul
                      className="uk-sidebar-nav-dropdown-items uk-nav-sub"
                      id={NavConstant.SUBITEMID}
                    >
                      {items._children && items._children.length > 0 &&
                        items._children.map((children: any, i: any) => {
                          return (
                            <Fragment key={i}>
                              <li
                                className="uk-sidebar-nav-item "
                                key={i.toString()}
                                id={NavConstant.SUBITEMID}
                                onClick={() => {handlesubItemClick(children,i)}}
                              >
                                <Link
                                  to={`${children.to}`}
                                  uk-tooltip={`title:${children.name}; pos:left;`}
                                  id={NavConstant.SUBITEMID}
                                  className={`uk-sidebar-nav-link  ${
                                    subchildpath === children.to
                                      ? "uk-active"
                                      : `${cssClassName.COLOR_WHITE}`
                                  }`}
                                >
                                  <i
                                    id={NavConstant.SUBITEMID}
                                    className={`${children.icon}`}
                                  ></i>
                                  <span id={NavConstant.SUBITEMID}>{children.name}</span>
                                </Link>
                              </li>
                            </Fragment>
                          );
                        })}
                    </ul>
                  )}
                </li>
              </Fragment>
            );
          })}
        </ul>
      </div>
    </>
  );
};

export default React.memo(TheSidebar);
