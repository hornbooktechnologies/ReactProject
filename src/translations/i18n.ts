import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
