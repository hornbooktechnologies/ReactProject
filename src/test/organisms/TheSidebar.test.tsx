import React from "react";
import renderer from "react-test-renderer";
import TheSidebar from "../../components/organisms/TheSidebar";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
import { BrowserRouter } from "react-router-dom";

describe("Component: TheSidebar", () => {
  it("TheSidebar renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <I18nextProvider i18n={i18n}>
            <TheSidebar />
          </I18nextProvider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
