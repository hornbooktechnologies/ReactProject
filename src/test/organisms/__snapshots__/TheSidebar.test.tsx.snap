// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TheSidebar TheSidebar renders correctly 1`] = `
<div
  className="uk-sidebar sidebar-left"
>
  <ul
    className="uk-sidebar-nav h-100  uk-ps"
    uk-scrollspy="closest: li; scroll: true"
  >
    <li
      className="uk-sidebar-nav-item uk-c-pointer uk-parent uk-sidebar-nav-dropdown uk-show"
      onClick={[Function]}
    >
      <a
        className="uk-sidebar-nav-link1"
        href="false"
        onClick={[Function]}
        tabIndex={0}
        uk-tooltip="title:NAVIGATION.OVERVIEW; pos:left;"
      >
        <i
          className="jera-home-trend-up"
        />
        <span>
          NAVIGATION.OVERVIEW
        </span>
        <i
          className="jera-arrow-down-1"
        />
      </a>
      <ul
        className="uk-sidebar-nav-dropdown-items uk-nav-sub"
        id="subItemId"
      >
        <li
          className="uk-sidebar-nav-item "
          id="subItemId"
          onClick={[Function]}
        >
          <a
            className="uk-sidebar-nav-link  uk-active"
            href="/CurrentStatus"
            id="subItemId"
            onClick={[Function]}
            uk-tooltip="title:NAVIGATION.CURRENT_STATUS; pos:left;"
          >
            <i
              className="jera-graph"
              id="subItemId"
            />
            <span
              id="subItemId"
            >
              NAVIGATION.CURRENT_STATUS
            </span>
          </a>
        </li>
        <li
          className="uk-sidebar-nav-item "
          id="subItemId"
          onClick={[Function]}
        >
          <a
            className="uk-sidebar-nav-link  color-white"
            href="/AnnualTarget"
            id="subItemId"
            onClick={[Function]}
            uk-tooltip="title:NAVIGATION.ANNUAL_TARGET; pos:left;"
          >
            <i
              className="jera-graph"
              id="subItemId"
            />
            <span
              id="subItemId"
            >
              NAVIGATION.ANNUAL_TARGET
            </span>
          </a>
        </li>
        <li
          className="uk-sidebar-nav-item "
          id="subItemId"
          onClick={[Function]}
        >
          <a
            className="uk-sidebar-nav-link  color-white"
            href="/OutageOutputlimit"
            id="subItemId"
            onClick={[Function]}
            uk-tooltip="title:NAVIGATION.OUTAGE_OUTPUT_LIMIT; pos:left;"
          >
            <i
              className="jera-status"
              id="subItemId"
            />
            <span
              id="subItemId"
            >
              NAVIGATION.OUTAGE_OUTPUT_LIMIT
            </span>
          </a>
        </li>
        <li
          className="uk-sidebar-nav-item "
          id="subItemId"
          onClick={[Function]}
        >
          <a
            className="uk-sidebar-nav-link  color-white"
            href="/UnitOperation"
            id="subItemId"
            onClick={[Function]}
            uk-tooltip="title:NAVIGATION.SPREAD_OPERATING; pos:left;"
          >
            <i
              className="jera-building-4"
              id="subItemId"
            />
            <span
              id="subItemId"
            >
              NAVIGATION.SPREAD_OPERATING
            </span>
          </a>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
